packages/toolkit/src/view/knowledge/components/notifications/InsufficientStorageBanner.tsx

```diff
@@ -1,18 +1,38 @@
 "use client";
 
 import { useRouter } from "next/navigation";
+import { Nullable } from "instill-sdk";
 
 import { Button, Icons, LinkButton } from "@instill-ai/design-system";
 
 type InsufficientStorageBannerProps = {
   setshowStorageWarning: React.Dispatch<React.SetStateAction<boolean>>;
+  plan: string;
+  namespaceType: Nullable<"user" | "organization">;
+  selectedNamespace: Nullable<string>;
 };
 
 export const InsufficientStorageBanner = ({
   setshowStorageWarning,
+  plan,
+  namespaceType,
+  selectedNamespace,
 }: InsufficientStorageBannerProps) => {
   const router = useRouter();
 
+  const getUpgradeLink = () => {
+    if (plan === "PLAN_FREE" && namespaceType === "user") {
+      return "/subscribe";
+    } else if (plan === "PLAN_FREE" && namespaceType === "organization") {
+      return `/${selectedNamespace}/organization-settings/billing/subscriptions/plan`;
```

Comment:
@thewbuk This is not correct. 

The namespace is id not the name, so you need to check what is its type using the useNamespaceType